packages/dds/tree/src/tableSchema.ts

```diff
@@ -961,17 +1039,19 @@ export namespace TableSchema {
 	 * @internal
 	 */
 	export interface SetCellParameters<
+		TCell extends ImplicitAllowedTypes = ImplicitAllowedTypes,
 		TColumn extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+		TRow extends ImplicitAllowedTypes = ImplicitAllowedTypes,
 	> {
 		/**
 		 * The key to uniquely identify a cell in a table.
 		 */
-		readonly key: CellKey;
+		readonly key: CellKey<TColumn, TRow>;
 
 		/**
 		 * The cell to set.
 		 */
-		readonly cell: InsertableTreeNodeFromImplicitAllowedTypes<TColumn>;
```

Comment:
This was a bug. We used the `TColumn` parameter to specify the cell type. So the behavior was correct, but the naming here was incorrect.